crates/solver/src/solver.rs

```diff
@@ -83,6 +87,35 @@ pub trait Solver: Send + Sync + 'static {
     fn name(&self) -> &str;
 }
 
+/// The result a given solver achieved in the auction
+pub enum AuctionResult {
+    /// Solution was valid and was ranked at the given place
+    /// Rank 1 means the solver won the competition
+    Ranked(usize),
+
+    /// Solution was invalid for some reason
+    Rejected(SolverRejectionReason),
+}
+
+/// Reason for why a solution may have been invalid
+pub enum SolverRejectionReason {
+    /// The solver didn't return a successful response in time
+    RunError(SolverRunError),
+
+    /// The solution candidate didn't include any user orders
+    NoUserOrders,
+
+    /// The solution candidate didn't include any mature user orders
+    NoMatureOrders,
+
+    /// The solution violated a price constraint (ie. max deviation to external price vector)
+    PriceViolation,
+
+    /// The solution didn't pass simulation. Includes revert reason as well as the calldata
+    /// to re-create simulation locally
+    SimulationFailure(String, Vec<u8>),
```

Comment:
instead of `String` it might make sense to send the `SimulationRequest` (https://github.com/cowprotocol/services/blob/0b3b8c8c01f82a13498166ea2b1f1514de3e11fc/crates/shared/src/code_simulation.rs#L111-L124) json.
Two reasons:
1) get parameters are limited in length
2) You might change how you do simulation in the future, perhaps adding overrides, and having the full json will be easier to repro.

Reply:
I think we should call you with a proper payload (not necessarily just get params). The reason I'm reluctant to send a Tenderly simulation request is that we don't necessarily want to dictate which tool solvers should use to evaluate their failures (they may as well use a vanilla node). I agree we should add all the relevant parts though that you might not know to reconstruct the exact transaction (such as gas, gas_price)